consensus/src/liveness/leader_reputation.rs

```diff
@@ -76,21 +76,7 @@ impl AptosDBBackend {
         // assumes target round is not too far from latest commit
         let limit = self.window_size + self.seek_len;
 
-        // there is a race condition between the next two lines, and new events being added.
-        // I.e. when latest_db_version is fetched, and get_events are called.
-        // if in between a new entry gets added max_returned_version will be larger than
-        // latest_db_version, and so we should take the max of the two.
-
-        // we cannot reorder those two functions, as if get_events is first,
-        // and then new entry gets added before get_latest_version is called,
-        // we would incorrectly think that we have a newer version.
-        let events = self.aptos_db.get_events(
-            &new_block_event_key(),
-            u64::max_value(),
-            Order::Descending,
-            limit as u64,
-            latest_db_version,
-        )?;
```

Comment:
This used to return events UP TO `latest_db_version` and the PR makes it ignore the restriction. 

However the comments being deleted here, and the fact that we are returning `max(latest_db_version, max_returned_version)` seem to indicate there was a misunderstanding about the `get_events(.., ledger_version)` semantics in the original code. 

Net-net, the original code never returns a version or any event newer than the passed-in `latest_db_version` if I read it correctly.. while in the new code it's possible to return things newer than `lastest_db_version`.

@igor-aptos  @zekun000 ? 

But anyway, the API on the db should take a `ledger_version` like other APIs.

Reply:
> seem to indicate there was a misunderstanding about the get_events(.., ledger_version) semantics in the original code.
actually the `ledger_version` parameter was added after the fact and resulted in the behavior that doesn't match the comment
https://github.com/aptos-labs/aptos-core/pull/2652

however that happened before launch and I'm not sure if it's okay to just change it without gating.  @igor-aptos ?

Reply:
Seems like comment is outdated, but I don't think this change is correct/equivalent.

we need limit events up to ledger_version. making this change - we might miss some, if latest is later than latest_db_version ?

Reply:
Offline discussed, this should be ok.